code/code/sys/indexdata.cc

```diff
@@ -18,167 +18,108 @@ extern FILE *mob_f;
+      max_exist(-99),
+      spec(0),
+      weight(0) {}
 
 indexData::indexData(const indexData &a) :
   virt(a.virt),
   pos(a.pos),
   number(a.number),
+  name(a.name),
+  short_desc(a.short_desc),
+  long_desc(a.long_desc),
+  description(a.description),
   max_exist(a.max_exist),
   spec(a.spec),
-  weight(a.weight)
-{
-  name = mud_str_dup(a.name);
-  short_desc = mud_str_dup(a.short_desc);
-  long_desc = mud_str_dup(a.long_desc);
-  description = mud_str_dup(a.description);
-}
-
-indexData::~indexData()
-{
-  delete [] name;
-  delete [] short_desc;
-  delete [] long_desc;
-  delete [] description;
-}
-
-mobIndexData::mobIndexData() :
-  faction(-99),
-  Class(-99),
-  level(-99),
-  race(-99),
-  doesLoad(false),
-  numberLoad(0)
-{
-}
-
-mobIndexData & mobIndexData::operator= (const mobIndexData &a)
-{
-  if (this == &a)
-    return *this;
-
-  indexData::operator=(a);
-
-  faction = a.faction;
-  Class = a.Class;
-  level = a.level;
-  race = a.race;
-  doesLoad = a.doesLoad;
-  numberLoad = a.numberLoad;
-
-  return *this;
-}
-
-mobIndexData::mobIndexData(const mobIndexData &a) :
-  indexData(a),
-  faction(a.faction),
-  Class(a.Class),
-  level(a.level),
-  race(a.race),
-  doesLoad(a.doesLoad),
-  numberLoad(a.numberLoad)
-{
-}
-
-mobIndexData::~mobIndexData()
-{
+  weight(a.weight) {}
+
+mobIndexData::mobIndexData()
+    : faction(-99),
+      Class(-99),
+      level(-99),
+      race(-99),
+      doesLoad(false),
+      numberLoad(0) {}
+
+mobIndexData::mobIndexData(int _virt, const sstring &_name, const sstring &_short_desc,
+                           const sstring &_long_desc, const sstring &_description, int _max_exist,
+                           int _spec, float _weight, long faction, long Class, long level,
+                           long race)
+    : faction(faction), Class(Class), level(level), race(race) {
+  virt = _virt;
+  name = _name;
+  short_desc = _short_desc;
+  long_desc = _long_desc;
+  description = _description;
+  max_exist = _max_exist;
+  spec = _spec;
+  weight = _weight;
 }
 
-objIndexData::objIndexData() :
-  ex_description(NULL),
-  max_struct(-99),
-  armor(-99),
-  where_worn(0),
-  itemtype(MAX_OBJ_TYPES),
-  value(-99)
-{
-}
-
-objIndexData & objIndexData::operator= (const objIndexData &a)
-{
-  if (this == &a)
-    return *this;
-
-  indexData::operator=(a);
-
-  // use copy operator;
-  if (a.ex_description)
-    ex_description = new extraDescription(*a.ex_description);
-  else
-    ex_description = NULL;
-
-  max_struct = a.max_struct;
-  armor = a.armor;
-  where_worn = a.where_worn;
-  itemtype = a.itemtype;
-  value = a.value;
-
-  int i;
-  for(i=0;i<MAX_OBJ_AFFECT;++i){
-    affected[i]=a.affected[i];
-  }
-
-  return *this;
+mobIndexData::mobIndexData(const mobIndexData &a)
+    : indexData(a),
+      faction(a.faction),
+      Class(a.Class),
+      level(a.level),
+      race(a.race),
+      doesLoad(a.doesLoad),
+      numberLoad(a.numberLoad) {}
+
+objIndexData::objIndexData()
```

Comment:
NSDMI

Reply:
Addressed in https://github.com/sneezymud/sneezymud/pull/380/commits/ca9f8b35b90fb1f6818a43d8dc48feadd8831a56